source/application/models/oxarticle.php

```diff
@@ -555,6 +555,9 @@ public function getActiveCheckQuery($blForceCoreTable = null)
         // check if article is still active
         $sQ = " $sTable.oxactive = 1 ";
 
+        // check if article is hidden
+        $sQ .= " and $sTable.oxhidden = 0 ";
```

Comment:
Maybe you schould change this to an OR clause for better usability.
I just see the case where someone wants to make an article visible to a collegue and thinks: "Okay it shouldn't be active for everyone in but hidden."
Maybe you should enclose this line with a check if the current user is an admin?